frontends/relay-futil/dahlia_functions.py

```diff
@@ -0,0 +1,139 @@
+import subprocess
+import os
+
+from tempfile import NamedTemporaryFile, TemporaryFile
+from futil_ast import *
+
+IMPORT_STATEMENT = """import "primitives/std.lib";\n"""
+NO_ERR = "2>/dev/null"
+
+
+def lower_dahlia_program(prog, component_name):
+    '''
+    Takes in a string representation of a Dahlia program, lowers it to FuTIL with the given `component_name`,
+    and applies the `externalize` pass. This pass exposes the inputs and outputs of primitive types that are
+    declared external, e.g. `std_mem_d1_ext`, and places them in the inputs and outputs of the respective component.
+
+    Example:
+        ------ Dahlia, component name: ProcessX ------
+        decl X: ubit<32>[4];
+        ...
+
+        ------------- Lower to FuTIL -----------------
+        component ProcessX() -> () {
+          X = prim std_mem_d1_ext(32, 4, 2);
+          ...
+        }
+
+        ------------- Externalize Pass ---------------
+        component ProcessX
+        (go: 1, clk: 1, X0_read_data: 32, X0_done: 1) ->
+        (done: 1, X0_addr0: 2, X0_write_data: 32, X0_write_en: 1, X0_clk: 1) {
+           ...
+        }
+
+    '''
+    program_string = ""
+    for line in prog.splitlines(): program_string += f'{line}\n'
+
+    with NamedTemporaryFile() as tf0, NamedTemporaryFile() as tf1, NamedTemporaryFile() as tf2:
+        tf0.write(bytes(program_string, 'UTF-8'))
+        tf0.seek(0)
+        tf1.seek(0)
+        tf2.seek(0)
+        fuse_binary = os.environ['DAHLIA_EXEC'] if 'DAHLIA_EXEC' in os.environ else 'fuse'
+        command = \
+            f"""
+            {fuse_binary} {tf0.name} --lower -b=futil -n={component_name} > {tf1.name} {NO_ERR} \
+            && cd ../../ && cargo run -- {tf1.name} -p externalize > {tf2.name} {NO_ERR} 
```

Comment:
don't need the `cd ../../`. `cargo run` works from any subdirectory of the futil root directly

Reply:
The issue I had with this is:
```
       46│+      Running `/Users/cgyurgyik/Projects/futil-clone/futil/target/debug/futil /var/folders/92/gt5cgc1d6rz4r1yvkl_p9www0000gn/T/tmpoq38bx18 -p externalize`
       47│+ Error: InvalidFile: Failed to read ./primitives/std.lib: No such file or directory (os error 2)
```
I think there is a way to pass in the path to the `std.lib`, but I don't recall how.

Reply:
you can use the `-l <path>` flag. but the thing to do here may be to just use `fud` directly. That `fud` can deal with finding the library for you. You can set the futil flags with `-s futil.flags '-p externalize'`.